### PR TITLE
fix/feat: Add and repair multiple converters for SD + other models

### PR DESCRIPTION
**Pull request by gs-olive**
# Description

- Focus on SD-performance-accelerating converters
  - Add minimal implementation for `broadcast_in_dim`
  - Add `prims.div` support
  - Fix bug in `sum`
  - Fix bug in `where`
- Add test cases for converters to avoid regressions

Fixes: #2346 

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Perf improvements

# Checklist:

- [ x ] My code follows the style guidelines of this project (You can use the linters)
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ x ] I have made corresponding changes to the documentation
- [ x ] I have added tests to verify my fix or my feature
- [ x ] New and existing unit tests pass locally with my changes
- [ x ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py [hunk 1]
```diff
@@ -1413,3 +1414,34 @@ def aten_ops_linear(
...
+        broadcast_node.args[1][i] == 1
+        for i in range(len(broadcast_node.args[1]))
+        if i not in broadcast_node.args[2]
+    )
+
+
+@dynamo_tensorrt_converter(
+    torch.ops.prims.broadcast_in_dim.default, capability_validator=broadcast_checker
+)  # type: ignore[misc]
+def aten_ops_broadcast_in_dim(
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:1432]: `prims_ops_broadcast_in_dim` --> in different file

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:1]: Alphabetize functions by name --> remove prim file

Comment (gs-olive) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:1]: Rename file `_ops_converters.py`

File: py/torch_tensorrt/dynamo/conversion/impl/reduce.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def sum(
     ):
         input_val = cast_trt_tensor(ctx, input_val, trt.float32, name)
 
-    if dim is None:
+    if dim is None or (isinstance(dim, (tuple, list)) and len(dim) == 0):
```
Comment (zewenli98) [inline py/torch_tensorrt/dynamo/conversion/impl/reduce.py:54]: Should we consider `dim` is [] or ()? In this case, the dim passed into axes will be 0.

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py [hunk 3]
```diff
@@ -1166,6 +1180,7 @@ def aten_ops_sub(
 @dynamo_tensorrt_converter(torch.ops.aten.div.Tensor_mode)  # type: ignore[misc]
 @dynamo_tensorrt_converter(torch.ops.aten.div.Scalar)  # type: ignore[misc]
 @dynamo_tensorrt_converter(torch.ops.aten.div.Scalar_mode)  # type: ignore[misc]
+@dynamo_tensorrt_converter(torch.ops.prims.div.default)  # type: ignore[misc]
```
Comment (zewenli98) [inline py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:1183]: Same as above.

**Review by zewenli98 — COMMENTED** [review]
Looks good overall. I was wondering if we add a prims decorator for those converters similar to aten, how should we specify source_ir? It doesn't seem appropriate to keep `SourceIR.ATEN`. Maybe write them again in prims_ops_converters? It feels sort of redundant...

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py [hunk 4]
```diff
@@ -651,14 +651,15 @@ def aten_ops_amax(
...
     target: Target,
     args: Tuple[Argument, ...],
     kwargs: Dict[str, Argument],
     name: str,
 ) -> Union[TRTTensor, Sequence[TRTTensor]]:
-    return impl.reduce.sum(
+    sum_ = impl.reduce.sum(
         ctx,
         target,
         SourceIR.ATEN,
```
[the remark this answers is not in the document]
Comment (gs-olive) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:665]: This is a good point - thanks for the catch. I will likely make a duplicate for now

File: py/torch_tensorrt/dynamo/conversion/impl/reduce.py (hunk 2 above)
Comment (gs-olive) [reply py/torch_tensorrt/dynamo/conversion/impl/reduce.py:54]: I thought that `dim` being empty implies the same thing as `dim` being None, which is to reduce over all axes

Comment (gs-olive) [reply py/torch_tensorrt/dynamo/conversion/impl/reduce.py:54]: This is the behavior in `torch`, at least with `torch.sum`

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py
Comment (gs-olive) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:1]: #2362

**Review by zewenli98 — APPROVED** [review]
@gs-olive Thanks! looks good to me!
